### DISCUSSION TITLE
Can't use locals defined in posthtml-expressions inside the layout

### DISCUSSION THREAD

**Discussion by thewebartisan7** [posthtml/posthtml#384] [Q&A] [answered]
It take me time to understand this, and I would like to know if this is how is designed or an issue.

Supposing we have defined locals via plugin expressions:

```js
              require("posthtml-expressions")({
                  locals: {
                    foo:  'bar'
                  },
              }),
```

I can use foo inside any pages but not inside a layout (a file that it's used to be extended using posthtml-extend.

I need to define again foo inside posthtml-extend like this:

```js
               require("posthtml-extend")({
                     locals: {
                        foo:  'bar'
                     },
                }),
```

Thanks

**Comment by cossssmin** [comment, marked as answer]
You need to pass `posthtml-expressions` as a plugin to `posthtml-extend`:

https://github.com/posthtml/posthtml-extend#plugins

**Reply by thewebartisan7** [reply]
Thank You! This make sense. I remember that I did this already but then I was not able to set removeScriptLocals. Having setup posthtml-expression separately it worked fine.

How to handle when I have several plugin that use expressions?
